plugin/src/main/java/org/opensearch/ml/settings/MLFeatureEnabledSetting.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.ml.settings;
+
+import static org.opensearch.ml.settings.MLCommonsSettings.ML_COMMONS_REMOTE_INFERENCE_ENABLED;
+
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+
+public class MLFeatureEnabledSetting {
+
+    private volatile Boolean isRemoteInferenceEnabled;
+
+    public MLFeatureEnabledSetting(ClusterService clusterService, Settings settings) {
+        isRemoteInferenceEnabled = ML_COMMONS_REMOTE_INFERENCE_ENABLED.get(settings);
+        clusterService
+            .getClusterSettings()
+            .addSettingsUpdateConsumer(ML_COMMONS_REMOTE_INFERENCE_ENABLED, it -> isRemoteInferenceEnabled = it);
+    }
+
+    /**
+     * Whether the remote inference feature is enabled.  If disabled, time series plugin rejects RESTful requests for this feature.
```

Comment:
`If disabled, time series plugin rejects RESTful requests for this feature.`  --> Fix this please